include/cpp-statsd-client/StatsdClient.hpp

```diff
@@ -116,33 +132,52 @@ inline void StatsdClient::send(const std::string& key,
                                const int value,
                                const std::string& type,
                                const float frequency) const noexcept {
-    const auto isFrequencyOne = [](const float frequency) noexcept {
-        constexpr float epsilon{0.0001f};
-        return std::fabs(frequency - 1.0f) < epsilon;
-    };
-
-    // Test if one should send or not, according to the frequency rate
-    if (!isFrequencyOne(frequency)) {
-        if (frequency < static_cast<float>(std::rand()) / RAND_MAX) {
-            return;
-        }
+    constexpr float epsilon{0.0001f};
+    const bool isFrequencyOne = std::fabs(frequency - 1.0f) < epsilon;
```

Comment:
frequency is constant, which means theres no point in evaluating if its `1.f` more than once. so it was just cleaner to remove the lambda and store the boolean to use over and over